python/python/lance/dataset.py

```diff
@@ -1103,7 +1103,7 @@ def cleanup_old_versions(
     def create_scalar_index(
         self,
         column: str,
-        index_type: Literal["BTREE"],
+        index_type: Union[Literal["BTREE"], Literal["BITMAP"]],
```

Comment:
Should we have a default index type to `BTREE`?

Reply:
We should automatically pick the index type based on cardinality by default.  I'm hoping to do this in a future PR (or maybe @raunaks13 )